chore: Move various conf files from /etc to /usr

HastD opened the pull request:
This avoids users editing the conf files and not receiving future updates to those files. All of these conf files can still be overridden with user-created conf files in `/etc`.

* Move distrobox.conf from `/etc/distrobox` to `/usr/share/distrobox`.
* Move service overrides, excluding the overrides that disable Xwayland, from `/etc/systemd` to `/usr/lib/systemd`, and rename the override files to be less generic and to avoid being overridden by `override.conf` files created by `systemctl edit`.
* Rename service overrides that disable Xwayland in GNOME and KDE to `99-noxwayland.conf` for clarity and consistency with Sway.
* Simplify some conditionals in `ujust install-vpn` using `case` blocks. (It was adjacent to code I was editing anyway to change the conf filenames.)

alexvojproc reviewed (approved):
looks good, we'll need to make sure the upcoming systemd service hardening overrides go into /usr too, unless they'll spend a trial period in /etc?

HastD commented:
@alexvojproc Yes, systemd service hardening done by secureblue should also go in `/usr/lib/systemd`. A "trial period" in `/etc/systemd` doesn't make sense to me—the distinction is whether the changes are made by the distribution or by the local admin. The local admin can override any such changes with their own drop-in files in `/etc/systemd` regardless.

RoyalOughtness commented:
> This avoids users editing the conf files and not receiving future updates to those files. 

It does do that but it also introduces two potentially new sources of headache:

- `ujust check-local-overrides` won't reflect user changes in /etc
- users can no longer simply delete our overrides from /etc

Also, if someone has already removed one of our overrides from /etc, will this change undo the effect of that?

HastD commented:
@RoyalOughtness Those are good points, let me go through those and see if I think it's still okay.

First, `ujust check-local-overrides` should still work fine: it shows files that users have *added* as well as existing files that users have modified. The difference is that the change will show up as a file listed as "Only in /etc/..." rather than as a diff on an existing file, but that doesn't seem like a problem.

Second, about users not being able to simply delete our overrides from `/etc`: I agree that this would be less than ideal for overrides that some users are likely to want to modify, since it makes overriding our overrides less convenient/intuitive (though still possible). Let's go through each of the files that this PR moves from `/etc` to `/usr`:

* `plasma-ksmserver.service.d/override.conf` and `plasma-xembedsniproxy.service.d/override.conf`: these just add an `ExecCondition` that prevents the services from running if (and *only* if) Xwayland is disabled. Since these services fail anyway when Xwayland is disabled, I can't think of any reason why someone would want to remove these.
* `rpm-ostreed-automatic.{service,timer}.d/override.conf`: You're right, there's a good chance some users will want to edit these; I'll move these two files back.
* `distrobox.conf`: I don't think anyone would want to delete the `LD_PRELOAD=` line as this simply prevents a bunch of spurious errors about not being able to preload `libhardened_malloc.so`. If users want to install and use hardened_malloc on an individual distrobox, they'd need to install it manually and can then simply edit `/etc/ld.so.preload` inside the distrobox. Also, user changes to `/etc/distrobox.conf` can still override settings from `/usr/share/distrobox/distrobox.conf`, so I don't think there's much risk of this one being a problem.

Finally, on this issue:

> Also, if someone has already removed one of our overrides from /etc, will this change undo the effect of that?

I hadn't thought of that; it would. As I said above, I don't think this will be a problem for the plasma and distrobox conf files, and I'll be moving back the rpm-ostreed-automatic conf files. Unfortunately, this would also be a problem for renaming the Xwayland overrides from `override.conf` to `99-noxwayland.conf`; I'll move those back as well to avoid this issue.

HastD committed:
chore: Move various conf files from /etc to /usr

This avoids users editing the conf files and not receiving future
updates to those files. All of these conf files can still be overridden
with user-created conf files in `/etc`.

* Move distrobox.conf from `/etc/distrobox` to `/usr/share/distrobox`.
* Move some service overrides from `/etc/systemd` to `/usr/lib/systemd`,
  and rename the override files to be less generic and to avoid being
  overridden by `override.conf` files created by `systemctl edit`.
* Simplify some conditionals in `ujust install-vpn` using `case` blocks.

Signed-off-by: Daniel Hast <hast.daniel@protonmail.com>

alexvojproc commented:
> `rpm-ostreed-automatic.{service,timer}.d/override.conf`: You're right, there's a good chance some users will want to edit these; I'll move these two files back.

@HastD Will the criterion for whether something goes in /etc/ or /usr/ just be "are users likely to want to modify it"? It still seems more elegant to me to put them in /usr/ and expect users to mask or modify them in /etc/.

> Also, if someone has already removed one of our overrides from /etc, will this change undo the effect of that?

I hate to get a reputation as "the person who breaks things", but is a breaking change not justified here? We were kind of objectively doing it wrong before 😅

HastD commented:
@alexvojproc Putting the `rpm-ostreed-automatic` overrides in `/usr` instead of `/etc` introduces a footgun for users: if they add an `OnCalendar` line to the timer with their preferred update timing without first adding an `OnCalendar=` line, it will add a new calendar trigger for the service *without* erasing the existing calendar trigger, which is a bit counterintuitive if you're not familiar with systemd unit configuration. I think it's simpler to keep the override in `/etc` so users can edit it directly.

RoyalOughtness commented:
> I hate to get a reputation as "the person who breaks things", but is a breaking change not justified here? We were kind of objectively doing it wrong before 😅

IMO this is a bit of a stretch :sweat_smile: 

We are overriding Fedora's changes in a way such that the user can remove the hardening and return to Fedora's settings.

RoyalOughtness commented:
Also, unintentionally breaking things is different from intentionally breaking things. In this case I'd rather keep doing things the "wrong way" but not introduce breakage, than change things around but break things.

RoyalOughtness commented:
Also, @HastD it would be good if this PR was broken up into smaller related chunks so that we can granularly rollback if needed.

HastD commented:
@RoyalOughtness Sure, will do. Multiple small PRs incoming. :sweat_smile:
